src/mixins/filter-mixin.js

```diff
@@ -15,15 +15,52 @@ export function addFilterHandler(filters, filter) {
   return filters
 }
 
-export function hasFilterHandler(filters, filter) {
-  if (typeof filter === "undefined") {
+/**
+ * hasFilterHandler
+ * - if testValue is undefined, checks to see if the chart has any active filters
+ * - if testValue is defined, checks to see if that testValue passes the active filter
+ *
+ * @param {*} filters - the chart's current filter
+ * @param {*} testValue - a value being tested to see if it passes the filter
+ *
+ *  - If chart values are not binned:
+ *      - Params will most likely both be an array of values
+ *      - e.g. [4,22,100] - and values 4, 22, and 100 will be selected in table chart
+ *        with all other values deselected
+ *  - If chart values are binned, params will most likely both be an array of arrays
+ *      - Inner arrays will represent a range of values
+ *      - e.g. [[19,27]] - the bin with values from 19 to 27 will be selected, with
+ *        all others being deselected
+ */
+export function hasFilterHandler(filters, testValue) {
+  if (typeof testValue === "undefined") {
     return filters.length > 0
-  } else if (Array.isArray(filter)) {
-    filter = filter.map(normalizeArrayByValue)
-    return filters.some(f => filter <= f && filter >= f)
-  } else {
-    return filters.some(f => filter <= f && filter >= f)
   }
+  testValue = Array.isArray(testValue)
+    ? testValue.map(normalizeArrayByValue)
+    : testValue
+  const filtersWithIsoDates = convertAllDatesToISOString(filters)
+  const testValueWithISODates = convertAllDatesToISOString(testValue)
+  return filtersWithIsoDates.some(
+    f => testValueWithISODates <= f && testValueWithISODates >= f
+  )
+}
+
+function convertAllDatesToISOString(a) {
+  if (Array.isArray(a)) {
+    // Assuming array can only have up to one level of nested arrays
+    return a.map(value => {
+      if (Array.isArray(value)) {
+        return value.map(v => {
+          if (v instanceof Date) {
+            return v.toISOString()
+          }
+        })
+      }
+      return value instanceof Date ? value.toISOString() : value
+    })
+  }
+  return a instanceof Date ? a.toISOString() : a
 }
```

Comment:
I'm pretty sure there's a logic bug in this function.

convertAllDatesToISOString will return the original values if they're in the top level array.

That is: convertAllDatesToISOString(['a', 'b', 'c']) will return ['a', 'b', 'c'], per the conditional on line 60.

But if you have a nested array of non-date values, it'll return undefined:

That is: convertAllDatesToISOString([['a', 'b', 'c']]) will return [undefined, undefined, undefined], because they won't be caught by the instanceof call on 55, so undefs are returned.

I assume that the conditional on lines 55-57 should just be replaced with the ternary on line 60. Maybe that could be abstracted out further, but meh.

Reply:
There definitely is. Fixed with recursion

Reply:
much nicer!